Talk-of-the-network.ipynb

```diff
@@ -0,0 +1,869 @@
+    "using GLM"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "MersenneTwister(UInt32[0x01332bfa], Base.dSFMT.DSFMT_state(Int32[-1772545288, 1073534108, 1077066014, 1072915095, -2146195133, 1072843413, 301764553, 1073404181, 750472136, 1073628106  …  -1491411563, 1073194977, 716119449, 1072893711, 1632331784, 758890923, 1433693833, -13012230, 382, 0]), [1.61032, 1.94373, 1.59225, 1.2466, 1.27962, 1.38458, 1.26211, 1.10108, 1.35191, 1.27754  …  1.48711, 1.53164, 1.84171, 1.54119, 1.59737, 1.13554, 1.14808, 1.90278, 1.99924, 1.55963], 382)"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "srand(20130810)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 1. Introduction \n",
+    "\n",
+    "In [Talk of the Network](https://www0.gsb.columbia.edu/mygsb/faculty/research/pubfiles/3391/TalkofNetworks.pdf), the authors  explore the pattern of personal communication betwee an individual's core friends group (strong ties) and a wider set of acquaintances (weak ties). This remarkable study is one of the first ones in marketing that explored the influence of social networks on the diffusion of marketing messages. The key questions investigated in the context of information dissemination are:\n",
+    "\n",
+    "- What matters more - strong ties or weak ties?\n",
+    "- What effect does the size of an average individuals network have?\n",
+    "- How does advertising interact with the diffusion through weak ties and that through strong ties\n",
+    "\n",
+    "In this workbook, we focus on the first question: do strong ties or weak ties influence the speed of information dissemination in a network?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2. Initializing the network"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since this study employs a set of synthetic networks, where each of the nodes have a fixed number of strong ties ($s$) and weak ties ($w$), we need to use the `MetaGraph` types to build these networks. We initialize the network as an empty graph and then build the neighborhoods of individual nodes by adhering to the number of strong and weak ties for each node."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "initialize_network (generic function with 1 method)"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "function initialize_network(n_nodes::Int, n_strong_ties::Int, n_weak_ties::Int)\n",
+    "    \n",
+    "    # Initialize an empty network\n",
+    "    \n",
+    "    mg = MetaGraph()\n",
```

Comment:
You're creating a zero-vertex, zero-edge metagraph but then setting vertex properties. This is supported (for now?), of course, but it's generally a bad idea. It also means that you don't have a proper graph structure.

Instead, `mg = MetaGraph(n_nodes)` to create a non-empty MetaGraph. (You're still not adding any edges, but that's a separate issue.)